ONSdigital/eq-author
# Add /surveys endpoint to the API

### What is the context of this PR?
Adds a /surveys endpoint to the API

### How to review 
Use the API to create surveys and associated schemas.

- Create a survey
```
POST /api/v1/surveys
{
   "survey_id": "001",
   "title": "My first survey",
   "questionnaires": []
}
```
- Add a schema that references the same `survey_id`
- Check that the schema is listed in the survey
```
GET /api/v1/surveys
```
- Get an individual survey
```
GET /api/v1/surveys/001
```



### Who worked on the PR

Describe who worked on the changes, so that other people can review.

This pull request adds a /surveys endpoint providing a mechanism for the
grouping of related questionnaires into a logical grouping. A
questionnaire now can't be created via the /schema endpoint without
creating a survey first. Once a survey is created with a given
survey_id, that survey_id can be used in requests to the /schema
endpoint.
